src/main/scala/com/gd/twitteranalytics/TransformTweets.scala

```diff
@@ -0,0 +1,58 @@
+package com.gd.twitteranalytics
+
+import java.util.{Date => JavaDate}
+import java.sql.Timestamp
+
+import TweetsDataSave.saveOutputToHdfs
+import org.apache.spark.SparkConf
+import org.apache.spark.streaming.dstream.DStream
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.functions.date_format
+import twitter4j.Status
+import com.gd.twitteranalytics.util._
+import org.apache.spark.rdd.RDD
+
+@SerialVersionUID(112L)
+object TransformTweets extends Serializable {
```

Comment:
it's now responsible not only for transformation, but for persistence of data. 
this is an incoherent design.

Reply:
Yes, I had this same doubt actually.
So here I have this function "processTweetsInfo" which is applied over a dStream and with in a forEachRdd, it does these 3 jobs:
* Convert RDD to DF (Tweets Transformation)
* Change the format of Date (Tweets Transformation)
* Save the Output to HDFS (Tweets Persistence)

So, all the above is happening inside a for each, how do I refactor it ..
 I created one TweetsDataSave also, but can't use it from driver.

Reply:
> how do I refactor it ..

there should be a different object per each role (extract, transform, load), and a Driver program should serve a role of a controller that combines the objects into a workflow.

> TweetsDataSave ..  can't use it from driver.

what's the problem with this one?

Reply:
Refactored the code. Moved processing of tweets to Driver.